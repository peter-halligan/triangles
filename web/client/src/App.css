.App {
  text-align: center;
  background-color: #282c34;
  color: white;
  min-height: 100vh;
}

.header {
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.form-wrapper {
  background-color: #282c34;
  color: white;
  padding: 3px;
  margin: 2px;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;

}

label {
  padding: 5px;
  margin: 3px;
}

input[type="text"] {
  width: 50px;
  border: none;
  background-color: white;
  padding: 5px;
  margin: 3px;
}
input[type="submit"] {
  background-color: #4CAF50; /* Green */
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}
.App-link {
  color: #61dafb;
}

