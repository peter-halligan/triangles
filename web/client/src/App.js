import React, { Component } from 'react';
import './App.css';
import { InputLengths } from "./components/InputLengths.compnent";
import { DisplayTriangleType } from "./components/DisplayTriangleType.component";
import { Header} from "./components/Header.component";
import {Instructions} from "./components/Instructions.component";

class App extends Component {
  constructor(props){
    super(props);
    this.state ={
      type: ''
    };
    this.setTriangleType = this.setTriangleType.bind(this);
  }

  setTriangleType(type){
    this.setState({...type})
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Instructions />
        <InputLengths setTriangleType={this.setTriangleType}/>
        {this.state.type && <DisplayTriangleType type={this.state.type}/>}
      </div>
    );
  }
}

export default App;
