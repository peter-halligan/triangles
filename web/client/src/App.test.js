import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test(' matches the snapshot ',  () =>{
  const tree = renderer
    .create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
  let instance = tree.getInstance();

  instance.setTriangleType({type: "test"})

  expect(instance.state).toEqual({type: 'test'});
  expect(tree.toJSON()).toMatchSnapshot();


});