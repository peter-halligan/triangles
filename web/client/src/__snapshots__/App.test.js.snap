// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` matches the snapshot  1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <h1>
      Triangle Type Checker
    </h1>
  </div>
  <div>
    <div>
      A Simple application to take in triangle side lengths and return the type of triangle.
    </div>
    <div>
      
    </div>
    <div>
      You can also download a cli which contains this functionality or just use the api try using post man to hit post /triangle {a:1, b:2 c: 3} 
    </div>
  </div>
  <div
    className="form-wrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Side A:
        <input
          name="a"
          onChange={[Function]}
          onFocus={[Function]}
          pattern="[0-9]*\\\\.{0,1}[0-9]"
          type="text"
          value=""
        />
      </label>
      <label>
        Side B:
        <input
          name="b"
          onChange={[Function]}
          onFocus={[Function]}
          pattern="[0-9]*\\\\.{0,1}[0-9]"
          type="text"
          value=""
        />
      </label>
      <label>
        Side C:
        <input
          name="c"
          onChange={[Function]}
          onFocus={[Function]}
          pattern="[0-9]*\\\\.{0,1}[0-9]"
          type="text"
          value=""
        />
      </label>
      <input
        type="submit"
        value="Check Triangle"
      />
    </form>
  </div>
  
</div>
`;

exports[` matches the snapshot  2`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <h1>
      Triangle Type Checker
    </h1>
  </div>
  <div>
    <div>
      A Simple application to take in triangle side lengths and return the type of triangle.
    </div>
    <div>
      
    </div>
    <div>
      You can also download a cli which contains this functionality or just use the api try using post man to hit post /triangle {a:1, b:2 c: 3} 
    </div>
  </div>
  <div
    className="form-wrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Side A:
        <input
          name="a"
          onChange={[Function]}
          onFocus={[Function]}
          pattern="[0-9]*\\\\.{0,1}[0-9]"
          type="text"
          value=""
        />
      </label>
      <label>
        Side B:
        <input
          name="b"
          onChange={[Function]}
          onFocus={[Function]}
          pattern="[0-9]*\\\\.{0,1}[0-9]"
          type="text"
          value=""
        />
      </label>
      <label>
        Side C:
        <input
          name="c"
          onChange={[Function]}
          onFocus={[Function]}
          pattern="[0-9]*\\\\.{0,1}[0-9]"
          type="text"
          value=""
        />
      </label>
      <input
        type="submit"
        value="Check Triangle"
      />
    </form>
  </div>
  <p>
    test
  </p>
</div>
`;
